### PR TITLE
 Redesign group list in the user settings view

### LINKED ISSUE #29057: Group members still displayed after group deletion
### Steps to reproduce
1. on the users page, add a group "abc"
2. put a user in the group
3. click on the group to display the users in that group
4. click the trashbin icon to delete the group

### Expected behaviour
Either:
1) the users listed disappear or
2) the group selection re-focuses to the group above, or the everyone group or something
and
3) the group no longer appears in the dropdown list of groups

### Actual behaviour
The list of users in that group remains displayed.
(note: refreshing the page confirms that the group and its memberships have actually been deleted, so this is just a UI refresh issue)

The group still appears in the dropdown list of groups that a user can be joined to (for creating a new user or updating an existing user)

### Server configuration
**Operating system**:
Ubuntu 16.04

**Web server:**
PHP dev server in dev environment

**Database:**
mySQL

**PHP version:**
v7.0

**ownCloud version:** (see ownCloud admin page)
10.0.3.3

using git master code in a development environment

### LINKED ISSUE #36273: Deleting groups not propagated to drop-down for creating users
### Steps to reproduce
1. Open user management screen.
2. Delete existing group.
3. Open drop-down that is used to add new (not yet created) users to a group

### Expected behaviour
The deleted group is not listed in the drop-down.

### Actual behaviour
The group is still there.

### Server configuration
**ownCloud version:** ownCloud Enterprise Edition 10.2.1 (stable)

**Updated from an older ownCloud or fresh install:** fresh

**Where did you install ownCloud from:** Docker Hub

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
Enhancement: Redesign group list in the user settings view

Before this PR, the group list settings were confusing as the user needs to
click on a group first before the trash bin icon appears to delete a group.
As well the user count was nondescript.
With this PR, a redesign takes care of those issues and also fixes the problem
that while deleting a group in the group list,
it was still available in the user group list.
Furthermore, after deleting a group, the user will be redirected to the
'Everyone' group summary view.
Also fixes an issue where the 'active'-class on a newly created group
was duplicated.

## Related Issue

- Fixes https://github.com/owncloud/core/issues/36273
- Fixes https://github.com/owncloud/core/issues/29057

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

### Before
image

### After
image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by JanAckermann** [comment]
@pmaier1 @micbar  FYI

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiProxySmoke-8-5-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32548/169/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingIntUsers1-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32548/142/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliAppManagement-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32548/118/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiProxySmoke-8-2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32548/166/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiWebdavEtagPropagation2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32548/86/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToShares2-git-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32548/110/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIFfSmoke-3-2-firefox-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32548/160/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiOnSqlite-sqlite-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32548/173/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliExternalStorage-10.7.0-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32553/121/1

**Comment by phil-davis** [comment]
- delete a group that has members and click "Yes" to confirm
- the group is gone from the left panel of groups - good
- the members of the group are still listed in the main user list, as if they are still in the group
Group-displayed-after-delete

In the screen-shot I had just deleted group "m123"

"something" (tm) needs to happen after a group is deleted, to refresh the user list to "something". Either go back to listing "Everyone" or just empty the list completely or?

File: changelog/unreleased/39262 [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Enhancement: Redesign group list in the user settings view
+
+Before this PR, the group list settings were confusing as the users needs to
```
Comment (phil-davis) [inline changelog/unreleased/39262:3]:
```suggestion
Before this PR, the group list settings were confusing as the user needs to
```

File: changelog/unreleased/39262 [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+click on a group first before the trash bin icon appears to delete a group.
+As well the user count was nondescript.
+With this PR, a redesign take care of those issues and also fixes the problem
```
Comment (phil-davis) [inline changelog/unreleased/39262:6]:
```suggestion
With this PR, a redesign takes care of those issues and also fixes the problem
```

File: changelog/unreleased/39262 [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+while deleting a group in the group list,
```
Comment (phil-davis) [inline changelog/unreleased/39262:7]:
```suggestion
that while deleting a group in the group list,
```

**Review by phil-davis — COMMENTED** [review]
Just minor edits to the changelog, while you look at what tests are failing and need adjusting.

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiTrashbinRestore-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32557/83/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIProxySmoke-3-2-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32559/163/1

**Comment by JanAckermann** [comment]
Another issue popped up (was pre existing)
1. Create a new group
2. Create a second group
3. Hover over the title
4. The Popover is displaying the wrong name

**Comment by JanAckermann** [comment]
@phil-davis after deletion of a group, we now will be redirected to everyone

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIManageUsersGrps-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32577/130/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIManageUsersGrps-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/32580/130/1

**Comment by JanAckermann** [comment]
@phil-davis CI should be green now, ready to have a second look =)

File: tests/acceptance/features/bootstrap/WebUIUsersContext.php [hunk 4]
```diff
@@ -907,19 +907,4 @@ public function theUserCountOfGroupShouldDisplayUsersOnTheWebUI($group, $count)
...
-	 * @return void
-	 */
-	public function theUserCountOfGroupShouldNotBeDisplayedOnTheWebui($group) {
-		$count = $this->usersPage->getUserCountOfGroup($group);
-		Assert::assertNull(
-			$count,
-			"Failed asserting that user count of group $group is not displayed"
-		);
-	}
 }
```
Comment (JanAckermann) [inline tests/acceptance/features/bootstrap/WebUIUsersContext.php:910]: We don't need this anymore, as we show the correct amount even 0

**Review by phil-davis — APPROVED** [review]
I can't "break" it any more. I manually tried putting users into various groups, and as subadmin of groups, then deleting the group... and it works nicely.

**Pull request merged**